rust/src/treefile.rs

```diff
@@ -1730,6 +1731,32 @@ impl Treefile {
                 .and_then(|m| m.install.as_ref().map(|i| !i.is_empty()))
                 .unwrap_or_default()
     }
+
+    /// Derive RPM importer flags for a given package and treefile settings.
+    pub fn importer_flags(&self, pkg_name: &str) -> Box<RpmImporterFlags> {
```

Comment:
I'm not opposed to this as is but I'd been thinking of the importer as a *higher* layer than treefile.  Hmm. 

I think what we really want down the line is for the importer to hold a reference to the treefile or so?

That said right now, if you agree that importer is higher than treefile, this method could move to the importer, and take a treefile as input.

Reply:
Interesting point. I first encountered this code today, and the impression I got is that the Treefile is slicing a subset of its configuration (in a bitflags-based way, but that's a technical detail) to initialize the importer.

The current code mostly follows an "happens before" relationship where the chain is `Treefile -> ImporterConfigFlags -> Importer`.

I can't honestly say which entity should be the higher level here, but moving this to the importer can also be fine.
Given cxx-rs limitations, it would become a freestanding `rpm_importer_compute_flags(tf: &Treefile, pkg_name: &str) -> Box<RpmImporterFlags>`.
There is another initialization path which just uses the defaults, so possibly holding a Treefile reference in the importer is not a feasible path. Or maybe should the daemon-side importing logic also respect the treefile settings?

Reply:
I'm merging this as-is given that CI is green and it already got reviewed.
I agree that there is a larger question around Treefile<->RpmImporter relationship, and this method will likely move to a better place depending on the outcome of that conversation. I'm splitting that to a dedicated ticket at https://github.com/coreos/rpm-ostree/issues/3676.